OpenSim/Sandbox/ExampleHopperDevice/exampleHopperDevice.cpp

```diff
@@ -70,20 +70,51 @@ class Device : public ModelComponent {
     /** Add outputs so we can report device quantities we care about. */
     /** The length of the device from anchor to anchor point. */
     OpenSim_DECLARE_OUTPUT(length, double, getLength, SimTK::Stage::Position);
-
-    // TODO: add other outputs.
+    /** The lengthening speed of the device. */
+    OpenSim_DECLARE_OUTPUT(speed, double, getSpeed, SimTK::Stage::Velocity);
+    /** The force transmitted by the device. */
+    OpenSim_DECLARE_OUTPUT(tension, double, getTension, SimTK::Stage::Dynamics);
+    /** The power produced(+)/dissipated(-) by the device. */
+    OpenSim_DECLARE_OUTPUT(power, double, getPower, SimTK::Stage::Dynamics);
+    /** The height of the model that device is attached to. */
+    OpenSim_DECLARE_OUTPUT(height, double, getHeight, SimTK::Stage::Position);
+    /** The COM height of the model that device is attached to. */
+    OpenSim_DECLARE_OUTPUT(com_height, double, getCenterOfMassHeight,
+        SimTK::Stage::Position);
 
     /** Member functions to access values of interest from the device. */
     double getLength(const SimTK::State& s) const {
         return getComponent<PathActuator>("cableAtoB").getLength(s);
     }
+    double getSpeed(const SimTK::State& s) const {
+        return getComponent<PathActuator>("cableAtoB").getLengtheningSpeed(s);
+    }
+    double getTension(const SimTK::State& s) const {
+        return getComponent<PathActuator>("cableAtoB").computeActuation(s);
+    }
+    double getPower(const SimTK::State& s) const {
+        return getComponent<PathActuator>("cableAtoB").getPower(s);
+    }
+
+    //device can read model height as measured from the block to ground
```

Comment:
"as measured from the block to ground" -> "(the distance between the bottom of the lower block and the ground)"